backend/question-service/server/app/readme.md

```diff
@@ -0,0 +1,321 @@
+    GET /questions # Returns all questions (no filters)
+    ```
+
+    ```
+    GET /questions?difficulty=Easy&topic=Arrays # Returns all Easy questions with topic Arrays
+    ```
+
+    ```
+    GET /questions?difficulty=Easy&difficulty=Medium&topic=Hash%20Tables&topic=Arrays # Returns all Easy or Medium questions with topic Hash Tables or Arrays
+    ```
+
+    **Sample Response**:
+
+    ```json
+    {
+    "status": "success",
+    "data": {
+        "questions": [
+        {
+            "id": "1",
+            "title": "Find Maximum in Array",
+            ...
+        },
+        {
+            "id": "2",
+            "title": "Reverse Array",
+            ...
+        }
+        ]
+    }
+    }
+    ```
+
+2. `GET /questions/all` Get All Questions
+
+    - **Description**: Returns a list of all questions with all fields.
+
+    **Sample Request**:
+
+    ```
+    GET /questions/all
+    ```
+
+    **Sample Response**:
+
+    ```json
+    {
+    "status": "success",
+    "data": {
+        "questions": [
+        {
+            "id": "1",
+            "title": "Find Maximum in Array",
+            "difficulty": "Easy",
+            "topic_tags": ["Arrays"],
+            ...
+        },
+        {
+            "id": "2",
+            "title": "Reverse Array",
+            "difficulty": "Easy",
+            "topic_tags": ["Arrays"],
+            ...
+        }
+        ]
+    }
+    }
+    ```
+
+3. `GET /questions/{question_id}` Get a Single Question
+
+    - **Description**: Returns a single question matching the given id.
+    - **Path Parameters**:
+        - `question_id`: The integer ID of the question to retrieve. Required.
+
+    **Sample Request**:
+
+    ```
+    GET /questions/1
+    ```
+
+    **Sample Response**:
+
+    ```json
+    {
+    "status": "success",
+    "data": {
+        "question": {
+        "id": "1",
+        "title": "Find Maximum in Array",
+        "difficulty": "Easy",
+        "topic_tags": ["Arrays"],
+        ...
+        }
+    }
+    }
+    ```
+
+4. `GET /questions/filters` Get Question Filters
+
+    - **Description**: Returns a list of all possible filters for questions: Difficulty and Topic.
+
+    **Sample Request**:
+
+    ```
+    GET /questions/filters
+    ```
+
+    **Sample Response**:
+
+    ```json
+    {
+    "status": "success",
+    "data": {
+        "Easy": ["Arrays", "Strings"],
+        "Medium": ["Trees", "Graphs"],
+        "Hard": ["Dynamic Programming", "Recursion"]
+    }
+    }
+    ```
+
+### `/admin` Endpoints
+
+1. `POST /admin/questions` Create a New Question
+
+    - **Description**: Creates a new question in the database.
+    - **Request Body**:
+        - `question`: The question data in JSON format. **If no ID is provided, one will automatically be generated.**
+
+    **Sample Request**:
+
+    ```json
+    POST /admin/questions
+    {
+        "title": "New Question Title",
+        "difficulty": "Easy",
+        "topic_tags": ["Arrays"],
+        ...
+    }
```

Comment:
nit:
```suggestion
    {
        question: {
            "title": "New Question Title",
            "difficulty": "Easy",
            "topic_tags": ["Arrays"],
            ...
        }            
    }
```

what do you think of wrapping the request body with `question`? since that seems to be the pattern with the endpoints above

(same for PUT and DELETE endpoint request and response body shapes)

up to u ah if not worth the effort is fine also just an opinion

Reply:
I'm opting not to do this:
- Based on conversation with @sheimoria I might be adding question model as a check. 
- FastAPI has [request body typechecking](https://fastapi.tiangolo.com/tutorial/body/) so it makes sense to define Question as a class and use that as the request body